File: app/src/main/java/com/rel/csam/imagecatcher/ImageListAdapter.java
package com.rel.csam.imagecatcher;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

/**
 * Created by leechansaem on 2016. 11. 2..
 */
public class ImageListAdapter extends RecyclerView.Adapter<ImageListAdapter.ViewHolder> {
    private ArrayList<String> mImageList;
    private Context mContext;

    public ImageListAdapter(Context context,ArrayList<String> android) {
        this.mImageList = android;
        this.mContext = context;
    }

    public void setImageList(ArrayList<String> imageList) {
        this.mImageList = imageList;
    }

    @Override
    public ImageListAdapter.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.row, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ImageListAdapter.ViewHolder viewHolder, int i) {

//        viewHolder.title.setText(mImageList.get(i));
        Glide.with(mContext).load(mImageList.get(i)).thumbnail(0.1f).fitCenter().into(viewHolder.img);
    }

    @Override
    public int getItemCount() {
        if(mImageList != null) {
            return mImageList.size();
        } else {
            return 0;
        }
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        private TextView title;
        private CustomImageView img;
        public ViewHolder(View view) {
            super(view);

//            title = (TextView)view.findViewById(R.id.title);
            img = (CustomImageView) view.findViewById(R.id.img);
        }
    }
}

File: app/src/main/java/com/rel/csam/lab/ImageListAdapter.kt
package com.rel.csam.lab

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.bumptech.glide.Glide
import com.rel.csam.lab.model.LinkImage

import java.util.ArrayList

/**
 * Created by leechansaem on 2016. 11. 2..
 */
class ImageListAdapter(private val mContext: MainActivity, private var mImageList: ArrayList<LinkImage>) : RecyclerView.Adapter<ImageListAdapter.ViewHolder>() {

    fun setImageList(imageList: ArrayList<LinkImage>) {
        this.mImageList = imageList
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ImageListAdapter.ViewHolder {
        val view = LayoutInflater.from(viewGroup.context).inflate(R.layout.row, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ImageListAdapter.ViewHolder, i: Int) {
        val data = mImageList[i]
        //        viewHolder.title.setText(mImageList.get(i));
        viewHolder.img.setOnClickListener {
            if (data.url != null) {
                mContext.getImageToLink(data.url!!)
            }
        }
        Glide.with(mContext).load(data.image).thumbnail(0.5f).into(viewHolder.img)
    }

    override fun getItemCount(): Int {
        return mImageList.size
    }

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val title: TextView? = null
        val img: CustomImageView

        init {
            //            title = (TextView)view.findViewById(R.id.title);
            img = view.findViewById<View>(R.id.img) as CustomImageView
        }
    }
}
